### ISSUE TITLE
No way to force error on excessive column data

### ISSUE THREAD

**Issue by rex-remind101** [tilo/smarter_csv#284]
Actual output:
```
$cat cats.csv
first name,last name,dogs,cats,birds,fish
     Dan,McAllister,2,1,2,4,6
     Lucy,Laweless,,5,,
     Miles,O'Brian,,,,21
     Nancy,Homes,2,,1,

x = SmarterCSV.process('/Users/Me/Documents/parentsquare/parent_square/cats.csv', options: {verbose: true})
=> [{:first_name=>"Dan", :last_name=>"McAllister", :dogs=>2, :cats=>1, :birds=>2, :fish=>4},
 {:first_name=>"Lucy", :last_name=>"Laweless", :cats=>5},
 {:first_name=>"Miles", :last_name=>"O'Brian", :fish=>21},
 {:first_name=>"Nancy", :last_name=>"Homes", :dogs=>2, :birds=>1}]
```
Expected output:
```
SmarterCSV::ExcessiveColumnError line 1 has 1 additional column than there are column headers
````

**Comment by tilo** [comment]
thank you for bringing this up!
I haven't seen this type of malformed CSV before.

I'll think about how this can be best handled, while also providing a resilient parsing experience.

**Comment by tilo** [comment]
Should an error be raised?

Often times the consumer of a CSV file has no control over, or knowledge of how the CSV file was generated.

As a consumer, we want to be able to read all data if that is possible.

In this case, we could auto-repair the situation by generating the missing headers on-the-fly.
e.g. 

```
[
    {
           :first_name => "Dan",
            :last_name => "McAllister",
                 :dogs => 2,
                 :cats => 1,
                :birds => 2,
                 :fish => 4,
     :missing_header_1 => 6
    },
...
]```

**Comment by tilo** [comment]
@rex-remind101 this will be fixed in 1.13.0

* the default behavior will be to auto-generate column_N headers for the additional columns

* when option `:strict` is `true`, it will raise `SmarterCSV::MalformedCSV`
